src/core/string_set.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include <cstdint>
+#include <functional>
+#include <optional>
+
+#include "core/dense_set.h"
+
+extern "C" {
+#include "redis/sds.h"
+}
+
+namespace dfly {
+
+class StringSet : public DenseSet {
+ public:
+  uint64_t Hash(const void* ptr) const override;
+
+  bool Equal(const void* ptr1, const void* ptr2) const override;
+
+  size_t ObjectAllocSize(const void* s1) const override;
+
+  bool Add(std::string_view s1);
+
+  bool AddRaw(sds s1);
+
+  bool Erase(std::string_view s1);
+
+  bool EraseRaw(sds s1);
+
+  bool Contains(std::string_view s1) const;
+
+  bool ContainsRaw(sds s1) const;
+
+  void Clear() override;
+
+  std::optional<std::string> Pop();
+  sds PopRaw();
+
+  ~StringSet() {
+    Clear();
+  }
+
+  StringSet(std::pmr::memory_resource* res = std::pmr::get_default_resource()) : DenseSet(res) {
+  }
+
+  iterator<sds> begin() {
```

Comment:
I do not think you need to reimplement the same function in the derived class. it should come from the base class.
if it's not working, try adding "using DenseSet::begin;" statement inside the class.

Reply:
The templated functions are accessible without these definitions but as far as I can tell using a compressed for loop i.e `for (sds s : set)` requires the explicit definition of the specialized functions. I could be missing something though.

Reply:
ah, I missed that it's non-template.